x/auth/tx/decoder.go

```diff
@@ -79,3 +89,77 @@ func DefaultJSONTxDecoder(cdc codec.ProtoCodecMarshaler) sdk.TxDecoder {
 		}, nil
 	}
 }
+
+// rejectNonADR027 rejects txBytes that do not follow ADR-027. This function
+// only checks that:
+// - field numbers are in ascending order (1, 2, and potentially multiple 3s),
+// - and varints as as short as possible.
+// All other ADR-027 edge cases (e.g. TxRaw fields having default values) will
+// not happen with TxRaw.
+func rejectNonADR027(txBytes []byte) error {
```

Comment:
how about moving the new functions to a separate package? It seams it's not necessary tide to `auth/tx/decoder`

Reply:
It is actually and maybe should just be renamed

Reply:
Yeah this is not a generic ADR-027 checker, but a simpler version of it that checks it for txBytes